### ISSUE TITLE
Add option to specify method to send AWS Pinpoint events

### ISSUE THREAD

**Issue by Dilatorily** [aws-amplify/amplify-js#3042]
**Is your feature request related to a problem? Please describe.**
AWS Pinpoint users are currently using AWS Amplify to send analytics events through XHR POST requests. Some useful analytics metrics are the time spent on page, which would require an event to be sent when the browser is closed. However, most browsers will not send XHR POST requests when a page is closed. This makes AWS Amplify JS useless for those kind of analytics events.

**Describe the solution you'd like**
AWS Amplify could provide the option to configure the method used to send the analytics metrics through a callback function, and provide a default callback that uses the current POST XHR method. It would allow customers to customize how their AWS Pinpoint events are being sent.

**Describe alternatives you've considered**
Currently, customers has to either overload the XHR method, or to proxy every request to their own API.

**Additional context**
N/A

**Comment by roborourke** [comment]
Just about to open a similar request so +1.

In browsers that support it `Navigator.sendBeacon()` should be used to send events to Pinpoint to avoid the drawbacks of XHR.

@Dilatorily if it helps in the meantime I've implemented my own custom pageView event using `Analytics.record()` on a `beforeunload` event listener. It's annoying that there's a small delay in navigation but it's the only way I've managed to collect time on page and scroll depth metrics so far.

**Comment by iartemiev** [comment]
Hey @Dilatorily and @roborourke , we have published a fix for this that's currently in the `1.1.41-unstable.2` version of aws-amplify and will be part of the next release! Please let me know if this resolves the issue you were experiencing.

**Comment by Dilatorily** [comment]
@iartemiev I think #4012 only solves `_session.stop` events.
I'm also looking to use navigator.sendBeacon for other custom events when the page/browser is closed.

**Comment by iartemiev** [comment]
@Dilatorily you're correct, we're currently only using it for _session.stop. I misread your feature request. I will look into using it for custom events as well.
